sent_turtle.py

```diff
@@ -0,0 +1,101 @@
+class PostOffice:
+    """A Post Office class. Allows users to message each other.
+
+    :ivar int message_id: Incremental id of the last message sent.
+    :ivar dict boxes: Users' inboxes.
+
+    :param list usernames: Users for which we should create PO Boxes.
+    """
+
+    def __init__(self, usernames):
+        self.message_id = 0
+        self.boxes = {user: [] for user in usernames}
+
+    def send_message(self, sender, recipient,message_title, message_body, urgent=False):
+        """Send a message to a recipient.
+
+        :param str sender: The message sender's username.
+        :param str recipient: The message recipient's username.
+        :param str message_title: The title of the message.
+        :param str message_body: The body of the message.
+        :param urgent: The urgency of the message.
+        :type urgent: bool, optional
+        :return: The message ID, auto incremented number.
+        :rtype: int
+        :raises KeyError: if the recipient does not exist.
+        """
+        user_box = self.boxes[recipient]
+        self.message_id = self.message_id + 1
+        message_details = {
+            'id': self.message_id,
+            'title': message_title,
+            'body': message_body,
+            'sender': sender,
+            'read': False
+        }
+        if urgent:
+            user_box.insert(0, message_details)
+        else:
+            user_box.append(message_details)
+        return self.message_id
+
+    def read_inbox(self, username, number_of_messages=None):
+        """Returns first N number/all of messages in inbox
+
+
+        :param str username: The message sender's username.
+        :param int number_of_messages: Number of messages to return.
+        :return: N first numbers of messages in inbox.
+        :rtype: list
+        :raises KeyError: if the recipient's username does not exist.
+        """
+        messages = []
+        counter = 0
+        messages_to_read = number_of_messages
+        if number_of_messages is None:
+            messages_to_read = len(self.boxes[username])
+        for message in self.boxes[username]:
+            if counter < messages_to_read:
+                if not message['read']:
+                    message['read'] = True
+                    counter += 1
+                    messages.append(message)
+            else:
+                break
```

Comment:
PLease rewrite this using list comprehension,are all the condition statements necessary?